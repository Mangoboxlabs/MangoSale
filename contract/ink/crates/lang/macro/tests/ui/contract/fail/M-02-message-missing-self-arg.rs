use ink_lang as ink;

#[ink::contract]
mod missing_message_self_arg {
    #[ink(storage)]
    pub struct MissingMessageSelfArg {}

    impl MissingMessage {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn missing_self_arg() {}
    }
}

fn main() {}
