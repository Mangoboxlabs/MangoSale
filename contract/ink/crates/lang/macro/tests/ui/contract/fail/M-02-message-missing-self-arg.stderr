error: ink! messages must have `&self` or `&mut self` receiver
  --> $DIR/M-02-message-missing-self-arg.rs:14:9
   |
14 | /         #[ink(message)]
15 | |         pub fn missing_self_arg() {}
   | |____________________________________^
